pkg/endpoints/utils.go

```diff
@@ -107,3 +112,45 @@ func (r Resource) RegisterReadinessProbes(container *restful.Container) {
 
 	container.Add(wsv4)
 }
+
+// Store all types here that are reused throughout files
+type Extension struct {
+	Name string
+	Url  string
+	Port string
+}
+
+
+func (r Resource) RegisterExtension(container *restful.Container, namespace string) {
+	logging.Log.Info("Adding API for extensions")
+	svcs, err := r.K8sClient.CoreV1().Services(namespace).List(metav1.ListOptions{LabelSelector: "tekton-dashboard-extension=true"})
+	if err != nil {
+		logging.Log.Debug("no extension found")
+		return
+	}
+	ws := new(restful.WebService)
+	ws.
+		Path("/").
+		Consumes(restful.MIME_JSON).
+		Produces(restful.MIME_JSON)
+	for _, svc := range svcs.Items {
+		for key, url := range svc.ObjectMeta.Annotations{
+			if key == "tekton-dashboard-endpoints" {
+				logging.Log.Debugf("extension Url: %s", url)
+                                ext := Extension { Name: svc.ObjectMeta.Name, Url: url, Port: getPort(svc) }
+				ws.Route(ws.POST(url).To(ext.HandleExtension))
+			}
+		}
+	}
+	container.Add(ws)
+}
+
+func (ext Extension) HandleExtension(request *restful.Request, response *restful.Response) {
+	target, _ := url.Parse("http://" +  ext.Name + ":" + ext.Port + "/")
```

Comment:
Do we need a todo here for handling https eventually? 

Missing method comment